pkg/gui/gui.go

```diff
@@ -471,7 +508,22 @@ func (gui *Gui) Run() error {
 		return err
 	}
 
-	gui.goEvery(g, time.Second*60, gui.fetch)
+	if gui.Config.GetUserConfig().GetString("reporting") == "undetermined" {
+		gui.introAgree.Add(1)
+	}
+
+	go func() {
+		_, err := gui.fetch(g, g.CurrentView(), false)
```

Comment:
I'm concerned about the fact we're doing a fetch as soon as we start the application. Given that with slow internet speeds, fetches can be quite slow, we might end up with lots of users getting errors because of index.lock because they're trying to do git commands while other git commands are in process. Although admittedly I'm not fully across what causes index.lock errors and whether or not this would pose a real concern.

I've spent a bit of time working on a local dockerised git server that delays ever http request by 10 seconds to see what it looks like being on a very slow internet, but that's not fully done yet. I'll need to spend a bit more time testing to see if this initial fetch is an issue or not.

I do think though that we can change this logic to reduce unnecessary fetching. We only want to fetch for repos for which we don't yet know whether they are private. Once we do a fetch on any repo, private or public, we shouldn't need to do it again next time we start lazygit. Perhaps instead of a privateRepos array we in state we should have a repoInfo array where each element is something like
{
  path: /User/blah/blah/lazygit"
  private: true // or false or nil
}
and if on startup, the current repo's path isn't in that list, we do the fetch, then update the value of private to true or false, and then check for that on next startup so that we don't need to do the fetch. What are your thoughts on that?

Reply:
While i was chaining everything to that `{path ... , private ...}` i realized we don't need the whole saving of a private repo.  
The app can just check if the current working DIR is inside the recently opened repos and if so not show the popup and wait for 60 seconds before doing the first fetch